scripts/setup_test.py

```diff
@@ -342,6 +318,80 @@ def mock_exists(unused_path):
                 'https://github.com/yarnpkg/yarn/releases/download/'
                 'v1.17.3/yarn-v1.17.3.tar.gz'])
 
+    def test_package_install_with_windows_x86(self):
+        def mock_exists(unused_path):
+            return False
+        os_name_swap = self.swap(common, 'OS_NAME', 'Windows')
+        architecture_swap = self.swap(common, 'ARCHITECTURE', 'x86')
+        exists_swap = self.swap(os.path, 'exists', mock_exists)
+
+        with self.test_py_swap, self.create_swap, os_name_swap, exists_swap:
+            with self.download_swap, self.rename_swap, self.delete_swap:
+                with self.isfile_swap, architecture_swap:
+                    with self.download_node_on_windows_swap:
+                        setup.main(args=[])
+        check_function_calls = self.check_function_calls.copy()
+        del check_function_calls['recursive_chown_is_called']
+        del check_function_calls['recursive_chmod_is_called']
+        for _, item in check_function_calls.items():
+            self.assertTrue(item)
+        self.assertEqual(
+            self.urls, [
+                'https://nodejs.org/dist/v10.15.3/node-v10.15.3'
+                '-win-x86.zip',
+                'https://github.com/yarnpkg/yarn/releases/download/'
+                'v1.17.3/yarn-v1.17.3.tar.gz'])
+
+    def test_package_install_with_windows_x64(self):
+        def mock_exists(unused_path):
+            return False
+        os_name_swap = self.swap(common, 'OS_NAME', 'Windows')
+        architecture_swap = self.swap(common, 'ARCHITECTURE', 'x86_64')
+        exists_swap = self.swap(os.path, 'exists', mock_exists)
+
+        with self.test_py_swap, self.create_swap, os_name_swap, exists_swap:
+            with self.download_swap, self.rename_swap, self.delete_swap:
+                with self.isfile_swap, architecture_swap:
+                    with self.download_node_on_windows_swap:
+                        setup.main(args=[])
+        check_function_calls = self.check_function_calls.copy()
+        del check_function_calls['recursive_chown_is_called']
+        del check_function_calls['recursive_chmod_is_called']
+        for _, item in check_function_calls.items():
+            self.assertTrue(item)
+        self.assertEqual(
+            self.urls, [
+                'https://nodejs.org/dist/v10.15.3/node-v10.15.3'
+                '-win-x64.zip',
+                'https://github.com/yarnpkg/yarn/releases/download/'
+                'v1.17.3/yarn-v1.17.3.tar.gz'])
```

Comment:
Add a note in the where the node version is defined and note that the tests here need to be changed if there is a node or yarn upgrade.

Reply:
Forgot about this, I will use my old pattern - import the versions from `common.py` instead of hard code it.

Reply:
Updated